.github/workflows/ci.yml

```diff
@@ -87,6 +87,41 @@ jobs:
         if: steps.cache.outputs.cache-hit != 'true' && matrix.cache == 'true'
         run: rm -rf ~/.m2/repository/io/trino/trino-*
 
+  reproducible-build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # checkout all commits, as the build result depends on `git describe` equivalent
+          ref: |
+            ${{ github.event_name == 'repository_dispatch' &&
+                github.event.client_payload.pull_request.head.sha == github.event.client_payload.slash_command.args.named.sha &&
+                format('refs/pull/{0}/head', github.event.client_payload.pull_request.number) || '' }}
+      - uses: ./.github/actions/setup
+        with:
+          cache: false # cache not used, dependencies will be downloaded to `maven.repo.local`
+          cleanup-node: true # not enough space to fetch dependencies and build twice
+      - name: First build to a local repository
+        # ~/.m2/repository isn't used because we are using `maven.repo.local` to store all downloaded dependencies and generated artifacts
+        run: |
+          export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
+          $MAVEN clean install ${MAVEN_FAST_INSTALL} -Dmaven.repo.local=build_1
+      - name: Second build to a local repository
+        # ~/.m2/repository isn't used because we are using `maven.repo.local` to store all downloaded dependencies and generated artifacts
+        run: |
+          export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
+          $MAVEN clean install ${MAVEN_FAST_INSTALL} -Dmaven.repo.local=build_2
+      - name: Compare builds
+        run: |
+          diff \
+              `# Excluding maven metadata files - downloaded dependencies and generated artifacts should be exactly the same` \
+              --exclude=maven-metadata-local.xml \
+              --exclude=_remote.repositories \
+              --exclude="*.lastUpdated" \
+              --exclude="resolver-status.properties" \
+              -bur build_1/io/trino build_2/io/trino
```

Comment:
Move the `-bur` above, so that it's easier to see what this command is during. I also prefer to make them separate args for readability:
```shell
diff -r -u -b \
    --exclude ... \
    build_1/io/trino build_2/io/trino
```
Could also remove the duplication
```shell
{build_1,build2}/io/trino

Reply:
Why do we need the `-b` flag? What text files do we have, other than POMs? I assume the primary goal here is to compare the binary JAR files.